src/components/TodoList/TodoList.tsx

```
@@ -1 +1,39 @@
-export const TodoList = () => {};
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUserById } from '../../utils/getUser';
+import { TodoInfo } from '../TodoInfo';
+
+interface Props {
+  todos: Todo[];
+  users?: User[];
+}
+
+export const TodoList = ({ todos, users }: Props) => {
+  if (!todos) {
+    return null;
+  }
+
+  return (
+    <section className="TodoList">
+      {todos.map(todo => {
+        const existingUser = (todo as Todo & { user?: User }).user;
+
+        let user: User | undefined;
+
+        if (existingUser) {
+          user = existingUser;
+        } else if (users) {
+          user = getUserById(todo.userId, users);
+        }
+
+        if (!user) {
```

Inside the map the code falls back to `getUserById(todo.userId, users)` and may `return null` when no user is found. This continues to perform data preparation/lookup in the child and hides missing-data problems by silently returning `null`. This violates checklist item #5: "Data preparation should happen in one place (App component) and be passed to child components; do not import data in non-root components." Ensure each `todo` passed in already contains the `user` object so the list rendering never needs to look up or skip items.